src/services/formatting/formatting.ts

```diff
@@ -493,9 +480,9 @@ namespace ts.formatting {
                     }
                 },
                 getIndentation: () => indentation,
-                getDelta: () => delta,
+                getDelta: (child) => (!delta || SmartIndenter.shouldInheritParentIndentation(node, child)) ? 0 : delta,

```

Comment:
I'm a little confused about the condition here - I would think that you want to maintain the delta if you're inheriting from the parent.